Source/WebKit/UIProcess/API/Cocoa/_WKWebExtension.h

```diff
@@ -52,52 +65,144 @@ typedef NS_ERROR_ENUM(_WKWebExtensionErrorDomain, _WKWebExtensionError) {
     _WKWebExtensionErrorBackgroundContentFailedToLoad,
 } NS_SWIFT_NAME(_WKWebExtension.Error) WK_API_AVAILABLE(macos(WK_MAC_TBA), ios(WK_IOS_TBA));
 
+/*! @abstract This notification is sent whenever a @link WKWebExtension @/link has new errors or errors were cleared. */
 WK_API_AVAILABLE(macos(WK_MAC_TBA), ios(WK_IOS_TBA))
 WK_EXTERN NSNotificationName const _WKWebExtensionErrorsWereUpdatedNotification NS_SWIFT_NAME(_WKWebExtension.errorsWereUpdatedNotification);
 
+/*!
+ @abstract A `WKWebExtension` object encapsulates a web extension’s resources that are defined by a `manifest.json` file.
+ @discussion This class handles the reading and parsing of the manifest file along with the supporting resources like icons and localizations.
+ */
 WK_CLASS_AVAILABLE(macos(WK_MAC_TBA), ios(WK_IOS_TBA))
 @interface _WKWebExtension : NSObject
 
 + (instancetype)new NS_UNAVAILABLE;
 - (instancetype)init NS_UNAVAILABLE;
 
+/*!
+ @abstract Returns a web extension initialized with a specified app extension bundle.
+ @param bundle The bundle to use for the new web extension.
+ @result An initialized web extension, or nil if the object could not be initialized.
+ @discussion This is a designated initializer.
+ */
 - (instancetype)initWithAppExtensionBundle:(NSBundle *)appExtensionBundle NS_DESIGNATED_INITIALIZER;
+
+/*!
+ @abstract Returns a web extension initialized with a specified app extension bundle.
+ @param resourceBaseURL The directory URL to use for the new web extension.
+ @result An initialized web extension, or nil if the object could not be initialized.
+ @discussion This is a designated initializer. The URL must be a file URL that points
+ to a directory containing a `manifest.json` file.
+ */
 - (instancetype)initWithResourceBaseURL:(NSURL *)resourceBaseURL NS_DESIGNATED_INITIALIZER;
 
-@property (readonly, nullable, nonatomic) NSArray<NSError *> *errors;
+/*! @abstract The active errors for the extension. Returns `nil` if there are no errors. */
+@property (nonatomic, nullable, readonly, copy) NSArray<NSError *> *errors;
+
+/*! @abstract The parsed manifest as a dictionary. Returns `nil` if the manifest was unable to be parsed. */
+@property (nonatomic, nullable, readonly, copy) NSDictionary<NSString *, id> *manifest;
 
-@property (readonly, nullable, nonatomic) NSDictionary<NSString *, id> *manifest;
+/*! @abstract The parsed manifest version. */
+@property (nonatomic, readonly) double manifestVersion;
 
-@property (readonly, nonatomic) double manifestVersion;
+/*!
+ @abstract Checks if a maniferst version is supported by the extension.
+ @param manifestVersion The version number to check.
+ @result Returns `YES` if the extension specified a manifest version that is is greater than or equal to `manifestVersion`.
+ */
 - (BOOL)usesManifestVersion:(double)manifestVersion;
```

Comment:
I missed this in earlier review, but is there a reason for the manifestVersion to be a double instead of an int?

Reply:
Future proofing. There have been talkings in http://github.com/w3c/webextensions about maybe doing a 3.1 or dot versions in general.